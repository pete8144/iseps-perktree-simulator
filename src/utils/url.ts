import qs from 'qs'

import { MAX_SE, MIN_SE } from '../constants'

export const decodePerks = (code: string): string[] => {
  return []
}

export const encodePerks = (perks: string[]): string => {
  return ''
}

export const readFromQs = (): { se?: number; perks?: string[] } => {
  const params = qs.parse(window.location.search.substring(1))

  // @ts-ignore
  const se = parseInt(params.se, 10)

  if (Number.isInteger(se) && se >= MIN_SE && se <= MAX_SE) {
    return { se }
  }

  if (params.perks && typeof params.perks === 'string') {
    return { perks: decodePerks(params.perks) }
  }

  return {}
}

export const saveSeToQs = (se: number) => {
  if (Number.isInteger(se) && se >= MIN_SE && se <= MAX_SE) {
    window.history.replaceState(null, '', '?se=' + se)
  }
}

export const savePerksToQs = (perks: string[]) => {
  // const p = encodePerks(perks)
  // window.history.replaceState(null, '', '?perks=' + p)
  window.history.replaceState(null, '', '?')
}
