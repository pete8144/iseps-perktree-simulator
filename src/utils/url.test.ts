// @ts-nocheck

import * as url from './url'

test('encode perks', () => {
  expect(
    url.encodePerks([
      'R1',
      'R2',
      'R3',
      'R4',
      'R5',
      'R6',
      'R7',
      'B1',
      'B2',
      'B3',
      'B4',
      'B5',
      'B6',
      'B7',
      'Y1',
      'Y2',
      'Y3',
      'Y4',
      'Y5',
      'Y6',
      'Y7',
    ]),
  ).toBe('RBY')
  expect(
    url.encodePerks([
      'R1',
      'R2',
      'R3',
      'R4',
      'R5',
      'B1',
      'B2',
      'B3',
      'B4',
      'B5',
      'B6',
      'B7',
      'P1',
      'P3',
      'P4',
      'P5',
      'P6',
      'Bu1',
      'Bu2',
      'W1',
      'W3',
      'W4',
      'T1',
      'T2',
      'T3',
      'T5',
      'T6',
    ]),
  ).toBe('R1-5BP1,3-6W1,3,4Bu1,2T1-3,5,6')
})

test('decode perks', () => {
  expect(url.decodePerks('RBY')).toMatchSnapshot()
  expect(url.decodePerks('R1-5B1,3P1,3-6Bu1-3W1,3,4')).toMatchSnapshot()
  expect(url.decodePerks('W1,3,6,8-10Ip6')).toMatchSnapshot()
})
