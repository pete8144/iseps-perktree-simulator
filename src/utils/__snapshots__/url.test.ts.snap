// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode perks 1`] = `
Array [
  "R1",
  "R2",
  "R3",
  "R4",
  "R5",
  "R6",
  "R7",
  "B1",
  "B2",
  "B3",
  "B4",
  "B5",
  "B6",
  "B7",
  "Y1",
  "Y2",
  "Y3",
  "Y4",
  "Y5",
  "Y6",
  "Y7",
]
`;

exports[`decode perks 2`] = `
Array [
  "R1",
  "R2",
  "R3",
  "R4",
  "R5",
  "B1",
  "B3",
  "P1",
  "P3",
  "P4",
  "P5",
  "P6",
  "Bu1",
  "Bu2",
  "Bu3",
  "W1",
  "W3",
  "W4",
]
`;

exports[`decode perks 3`] = `
Array [
  "W1",
  "W3",
  "W6",
  "W8",
  "W9",
  "W10",
  "IP",
  "IP",
  "IP",
  "IP",
  "IP",
  "IP",
]
`;
